File: src/main/java/com/wikia/tibia/objects/LootItem.java
package com.wikia.tibia.objects;

import com.wikia.tibia.enums.Rarity;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
public class LootItem {

    private String itemName;
    private String amount;
    private Rarity rarity;
}

File: src/main/kotlin/com/wikia/tibia/objects/LootItem.kt
package com.wikia.tibia.objects

import com.wikia.tibia.enums.Rarity
import java.util.*

data class LootItem(
        val itemName: String,
        val amount: String? = null,
        val rarity: Rarity? = null
) {

    override fun equals(other: Any?): Boolean {
        return other is LootItem &&
                itemName == other.itemName
    }

    override fun hashCode(): Int {
        return Objects.hash(itemName)
    }

    companion object {

        fun fromName(itemName: String): LootItem {
            return LootItem(
                    itemName = itemName,
                    amount = null,
                    rarity = null
            )
        }
    }
}